contracts/Removal.sol

```diff
@@ -80,37 +110,87 @@ contract Removal is
     return removalId.unpackRemovalIdV0();
   }
 
+  // TODO do we want a batch version of this?
+  /**
+   * @notice Get the restriction schedule id (which is the removal's project id) for a given removal id.
+   */
+  function getProjectIdForRemoval(uint256 removalId)
+    public
+    view
+    returns (uint256)
+  {
+    return _removalIdToProjectId[removalId];
+  }
+
+  /**
+   * @notice Get the restriction schedule data for a given project id.
+   */
+  function getScheduleData(uint256 projectId)
+    public
+    view
+    returns (ScheduleData memory)
+  {
+    return _projectIdToScheduleData[projectId];
```

Comment:
Is there any way to get this from a removal ID so it doesnt require looking up a project ID before creating a subsequent call using the returned projectID?If the requests come in two separate RPC calls, there's no guarantee that the schedule data is going to be unchanged since the retrieving the projectId-- another reason I don't like how the project ID is decoupled from the removal ID

Reply:
we can't get it from the removal id itself unless we go back and completely change how the removal ids are encoded... we're out of space.  as for the concern about going from project id to schedule data in two separate RPC calls.... I don't totally understand the concern because the project id associated with a removal is static, so it doesn't matter when you retrieve it it's always the same, so there's no race condition situation or anything.  also, the project id associated with a removal is also a connection that exists off chain (the project id is literally the project datastore id -- or at least will be when we implement that and removals are child entities).  This mapping needs to exist on chain though so that schedule creation and schedule proceed routing for removals can happen atomically and on-chain during the course of listing and selling removals, respectively

Reply:
Can this accept a removalId and return the schedule instead by accessing the projectId in the removal -> project 
 mapping?

Reply:
We could add a schedule lookup by removal id and as well as by project id.

Reply:
I think I'll do that, because the place in which this method is used in the `rNori` contract doesn't actually know or care which removal id it's working with